### ISSUE TITLE
server disconnect and reconnect

### ISSUE THREAD

**Issue by Khundian** [lformella/xdcc-grabscher#47]
XG sits on 10 servers and 20+ channels, but somehow the same server disconnects and reconnects regulary. The server I am talking about is irc.abjects.net. When I connect with mIRC, I not get disconnected. Am I the only one having this issue?

**Comment by Khundian** [comment]
XG is connected to about 20 servers and 50 channels now, and It seems its not related to only 1 server. I can see several servers/channels disconnect and reconnect randomly, somehow XG doesn't manage to stay connected to all of them at once.. at least not for long.
I must admit that elasticsearch is keeping my server + node very busy, always high cpu load. But I don't think this has anything to do with this issue. Although updating the elasticsearch plugin to work with the latest version might lead to optimizations :)

If you want I can send you my sqlite database for troubleshooting

**Comment by lformella** [comment]
Please enable Error logging by putting this file https://gist.github.com/lformella/10551086 into your XG user folder. And look for messages like "WatchConnection() connection seems hanging ...".

I implemented a connection watcher because I got hanged up bot connections on my windows download box. The connections was still open, but not receiving any data. The watcher also monitors normal irc connections, so maybe it might be buggy.

**Comment by Khundian** [comment]
I cannot see this message you talk about, I saw the problem get worse when I join more servers/channels. Can you please tell me how I can log to a file, I can send you the log file :)

**Comment by lformella** [comment]
You can use the following config:

``` xml
<log4net>
  <root>
    <level value="Info" />
    <appender-ref ref="FileAppender" />
  </root>
  <appender name="FileAppender" type="log4net.Appender.FileAppender">
    <file value="log.txt" />
    <appendToFile value="true" />
    <layout type="log4net.Layout.PatternLayout">
      <conversionPattern value="%date{dd-MM-yyyy HH:mm:ss,fff} %5level [%2thread] %line:%logger.%message%n" />
    </layout>
  </appender>
</log4net>
```

I am not sure if you will receive some useful error messages, because I am using SmartIrc4net and this lib might have a bug. Unfortunately it is currently build without log4net - I will change this in the next release.

**Comment by Khundian** [comment]
as you predicted, I didn't get much useful information from the logfile, lots of error notifications from a couple of servers who keep disconnecting. Though there is one section at the end of the log that might be useful to you, I sent you the log in your mail.

**Comment by Freedos** [comment]
I noticed this behaviour too on XG 3.X but didn't had anny issues on XG 2.X. Mabe you could use an old SmartIrc4net version?

**Comment by lformella** [comment]
Smart4irc was introduced in XG 2.2.0 and I already located a bottleneck and implemented a fix for XG 3.3

**Comment by lformella** [comment]
Fixed in [v3.3.0.0](https://github.com/lformella/xdcc-grabscher/releases/tag/v3.3.0.0)
